### PR TITLE
Check collateralization ratio of newly created tokens, rather than new Position

### PR DESCRIPTION
**Pull request by ptare**
Issue #1042 

Signed-off-by: Prasad Tare <prasad.tare@gmail.com>

### CODE REVIEW

File: core/test/financial-templates/PricelessPositionManager.js [hunk 1]
```diff
@@ -374,11 +374,27 @@ contract("PricelessPositionManager", function(accounts) {
...
     // global collateralization ratio = (150 + 15 + 25) / (100 + 10 + 10) = 1.58333
     // To maintain 10 tokens, need at least 15.833 collateral => can withdraw from 25 down to 16 but not to 15.
     assert(await didContractThrow(pricelessPositionManager.withdraw({ rawValue: toWei("10") }, { from: other })));
     await pricelessPositionManager.withdraw({ rawValue: toWei("9") }, { from: other });
+
+    // GCR = (15 + 16) / (10 + 10) = 1.55.
+    // `sponsor` has a CR of 1.5.
+    // `other` has a CR of 1.6.
+    // `sponsor` can create new tokens that are individually above 1.55, without needing to bring their whole
+    // position up to 1.55. `other` can create new tokens below their current CR as long as the new tokens are above
```
Comment (chrismaree) [inline core/test/financial-templates/PricelessPositionManager.js]: +1 on this test comment. makes it clear

File: core/contracts/financial-templates/implementation/PricelessPositionManager.sol [hunk 2]
```diff
@@ -248,14 +248,15 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
         onlyPreExpiration()
         fees()
     {
+        require(_checkCollateralization(collateralAmount, numTokens));
```
Comment (mrice32) [inline core/contracts/financial-templates/implementation/PricelessPositionManager.sol]: With how you've designed the collateralization checks below, it seems pretty easy to add an `||` check for the new tokens and the total position. Curious to hear your thoughts now that we know more about how the implementation will look. Are there any disadvantages to the `||` check?

Comment (ptare) [reply core/contracts/financial-templates/implementation/PricelessPositionManager.sol]: Yep, the implementation would be as simple as to move this line down below line 261 and do:

```
require(_checkCollateralization(collateralAmount, numTokens) || _checkCollateralization(positionData));
```

I don't have a strong opinion either way. Seems simpler to just not allow it, one less case that we have to prove is correct. (E.g., the GCR will now decrease after creating tokens).

Comment (ptare) [reply core/contracts/financial-templates/implementation/PricelessPositionManager.sol]: Happy to make this change in a follow up if we think that's preferable.

**Pull request merged**